### PR TITLE
Remove Cache

### PR DESCRIPTION
**Pull request by Hanmac**
## Subject

I am targeting this branch, because it is BC break.

## Changelog

```markdown
### Removed
- Remove cache related code.
```

## To do

- [x] Update the tests;
- [x] Update the documentation;
- [ ] Add an upgrade note.

### CODE REVIEW

**Comment by eerison** [comment]
@Hanmac I guess you need to rebase your branch

**Comment by Hanmac** [comment]
> @Hanmac I guess you need to rebase your branch

i think i need to wait for this until the CI has merged 3.x into 4.x

**Comment by eerison** [comment]
> > @Hanmac I guess you need to rebase your branch
> 
> i think i need to wait for this until the CI has merged 3.x into 4.x

I thought that Vincent has merged it, my bad

**Comment by jordisala1991** [comment]
Now it is merged, please rebase.

File: docs/reference/advanced_configuration.rst
Comment (VincentLanglet) [inline docs/reference/advanced_configuration.rst]:
```suggestion
```

File: docs/reference/advanced_usage.rst
Comment (VincentLanglet) [inline docs/reference/advanced_usage.rst]:
```suggestion
```

File: docs/reference/installation.rst
Comment (VincentLanglet) [inline docs/reference/installation.rst]:
```suggestion
```

File: src/Cache/BlockEsiCache.php
Comment (VincentLanglet) [inline src/Cache/BlockEsiCache.php:41]: We need to add a deprecation on 3.x for all the classes removed.

`@deprecated` on the class, possibly trigger a deprecation in the constructor, and add an UPGRADE note.

File: src/DependencyInjection/Configuration.php
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php:147]: We need to trigger a deprecation for this config node on 3.x

File: src/DependencyInjection/Configuration.php
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php:322]: We need to trigger a deprecation for this node on 3.x

**Review by jordisala1991 — DISMISSED** [review]

File: src/DependencyInjection/Configuration.php
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php:382]: I assume we have to keep this option for now in to order to avoid a BC break without any deprecation.

The default value should be change to false to be consistent and the option will not be used and removed only on 5.x. 
On 4.x it will trigger a deprecation when set.

File: tests/App/config/sonata.yaml
Comment (VincentLanglet) [inline tests/App/config/sonata.yaml]:
```suggestion
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

**Comment by VincentLanglet** [comment]
@jordisala1991, just saw you approved.
Should we deprecate the class/config on 3.x as always ; or do we take a shortcut here ?

**Comment by jordisala1991** [comment]
Not sure if we need upgrade note, since everything was deprecated and we have a default note for deprecated code. Wdyt @VincentLanglet ?

File: src/DependencyInjection/Configuration.php
Comment (Hanmac) [reply src/DependencyInjection/Configuration.php:382]: we could deprecate it in 3x?

**Comment by VincentLanglet** [comment]
> Not sure if we need upgrade note, since everything was deprecated and we have a default note for deprecated code. Wdyt @VincentLanglet ?

We just didn't directly deprecated the removed class/config.

For the config, you can still use
```
cache: false
```
and others cache config without any warning.

For the classes, if someone is using `BlockEsiCache` or others classes, directly in his code, he'll get some surprise in 4.x.

I would say it doesn't cost us too much to add 
```
@deprecated since sonata-project/page-bundle version 3.x and will be removed in 4.0.
```
on classes, and call `setDeprecated` (like https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/DependencyInjection/Configuration.php#L166-L171) on config keys.

Do you mind doing a PR on 3.x @Hanmac ?

**Comment by Hanmac** [comment]
> > Not sure if we need upgrade note, since everything was deprecated and we have a default note for deprecated code. Wdyt @VincentLanglet ?
> 
> We just didn't directly deprecated the removed class/config.
> 
> For the config, you can still use
> 
> ```
> cache: false
> ```
> 
> and others cache config without any warning.
> 
> For the classes, if someone is using `BlockEsiCache` or others classes, directly in his code, he'll get some surprise in 4.x.
> 
> I would say it doesn't cost us too much to add
> 
> ```
> @deprecated since sonata-project/page-bundle version 3.x and will be removed in 4.0.
> ```
> 
> on classes, and call `setDeprecated` (like https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/DependencyInjection/Configuration.php#L166-L171) on config keys.
> 
> Do you mind doing a PR on 3.x @Hanmac ?

Deprecated the Cache classes, and the config options:
https://github.com/sonata-project/SonataPageBundle/pull/1435

**Comment by VincentLanglet** [comment]
Thanks, I merged 3.x into 4.x so there is just some conflicts to solve

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**
